taskcat/deployer.py

```diff
@@ -121,6 +125,18 @@ def set_input_path(self, input_path):
     def get_input_path(self):
         return self._input_path
 
+    def set_prod_key_prefix(self, source_key_prefix, input_path):
+        if source_key_prefix is not None:
+            self._prod_key_prefix = source_key_prefix.strip('/') + '/'
+        elif source_key_prefix is None and input_path is not None:
+            # Assume the key prefix we are looking to replace matches the repo name
+            # Split on slashes assuming slash is at the end, we would need element before last
+            repo = input_path.split('/')[-2]
```

Comment:
This may be dangerous as input_path could be . or some relative path. I would think that it's best to just not act when the source_key_prefix isn't provided and assume that it stays the same as the default already in the templates... or even better, it just uses the target_key_prefix (which is only required when uploading).

Reply:
Thats fair. I will move this logic as an assumption outside of taskcat and would be passed in if met.